### DISCUSSION TITLE
AIO on Synology doesn't start / Database container connection? (& Domain Check failed)

### DISCUSSION THREAD

**Discussion by Fred-DTV** [nextcloud/all-in-one#1797] [Questions] [answered]
Hey hey,

I have been trying to get AIO running on our Synology NAS (in Docker) for quite a long time now, which actually helped me to finally understand quite a few network and server related topics :)

Unfortunately I am now stuck at the very last step: starting of the containers.
The mastercontainer successfully created all of the other containers, but except for the Talk container, none of them will properly start.

I think I have narrowed it down to **3 potential problems** (but ofc I am open to other ideas):

1. The database container isn't starting as it should. Unfortunately I couldn't find a reason in the logs
https://pastebin.com/ptSy3Z22

2. The containers can't communicate with each other. They are however all on the same docker network bridge

3. I wasn't able to validate the domain and had to skip the validation process. I have a working reverse proxy setup (I have tested it by pointing it to other docker containers first) and used this configuration to start the containers:

```
sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 8080:8080 \
-e APACHE_PORT=11000 \
-e APACHE_IP_BINDING=127.0.0.1 \
-e SKIP_DOMAIN_VALIDATION=true \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /volume1/docker/docker.sock:/var/run/docker.sock:ro \
-e NEXTCLOUD_DATADIR="/volume1/docker/nextcloud/data" \
nextcloud/all-in-one:latest

```
I have tried pointing the reverse proxy to port 11000 as well as to port 443 (with opening the port at the docker container) and both times the validation failed. I don't know if this could keep the containers from starting, though I am getting this error in the nextcloud container log: nc: getaddrinfo: Name does not resolve

### Host OS 
Synology DSM 7

#### Nextcloud AIO version 
5.1.0

#### Current channel 
latest

#### Other valuable info 
General notes for the installation process on a Synology NAS which I have encounterd on my way here: (maybe additions to readme?)
1. Starting the master container through the Docker GUI doesn't work because the ENV variables wouldn't be accepted. So one has to do this either through SSH or just by sending commands through user defined scripts as root
2.  I had to create a symlink for the docker.sock to avoid getting "Docker socket not available" errors even with binding the volume. This is the script I have used (and scheduled it daily in case of restarts or updates) `sudo ln -s /var/run/docker.sock /volume1/docker/docker.sock`

**Comment by Fred-DTV** [comment]
P.S.: in the reverse proxy readme for Synology is a typo: https://user-images.githubusercontent.com/89748315/192525681-c06f3b39-f510-458e-b1f2-6b2cd995e24c.png
$proxy_add_x_forwardeT_for instead of $proxy_add_x_forwarded_for

This is mine:
![Screenshot 2023-01-18 at 15 13 31](https://user-images.githubusercontent.com/70434961/213193789-fa936edc-e307-4e6a-9a53-ae26d1bf2f42.jpg)

**Reply by szaimen** [reply]
thanks for the screenshot! Am I allowd to exchange the screenshot with yours?

**Reply by Fred-DTV** [reply]
absolutely :)

As soon as the Nextcloud is running I could also provide you some screenshots for explaining how to setup the DNS Server on the Synology NAS to be able to access it from the local network through the domain too

**Reply by szaimen** [reply]
Thanks! :)

**Reply by szaimen** [reply]
updated

**Comment by szaimen** [comment]
> * The database container isn't starting as it should. Unfortunately I couldn't find a reason in the logs
>   https://pastebin.com/ptSy3Z22
> * The containers can't communicate with each other. They are however all on the same docker network bridge
> * I wasn't able to validate the domain and had to skip the validation process. I have a working reverse proxy setup (I have tested it by pointing it to other docker containers first) and used this configuration to start the containers:

Can you check if disabling your synology firewall makes things work? 

> General notes for the installation process on a Synology NAS which I have encounterd on my way here: (maybe additions to readme?)

Yes, a PR is welcome! Maybe add a section above https://github.com/nextcloud/all-in-one#how-to-run-aio-with-portainer.

> 1. Starting the master container through the Docker GUI doesn't work because the ENV variables wouldn't be accepted. So one has to do this either through SSH or just by sending commands through user defined scripts as root

Indeed, could be added to the doc.

> 2. I had to create a symlink for the docker.sock to avoid getting "Docker socket not available" errors even with binding the volume. This is the script I have used (and scheduled it daily in case of restarts or updates) `sudo ln -s /var/run/docker.sock /volume1/docker/docker.sock`

I dont understand. Why should mounting `/var/run/docker.sock ` into the container not work?

**Reply by Fred-DTV** [reply]
I don't know why I needed the symlink, but I had two different errors, depending on the setup, at first when trying to start the mastercontainer:
"docker socket is not available" or "can't connect to docker socket"

After creating the symlink, I haven't had this problem anymore, however, I was trying out other settings constantly too, so maybe something else solved it

**Reply by szaimen** [reply]
Then simply use 

`--volume /volume1/docker/docker.sock:/var/run/docker.sock:ro`
And `-e DOCKER_SOCKET_PATH=/volume1/docker/docker.sock`

**Reply by Fred-DTV** [reply]
ok fair enough 🙈 just restarted everything with these and it works like a charm

**Comment by Fred-DTV** [comment]
Hi @szaimen 

I can't say thank you enough! As soon as I turned off the firewall the domaincheck worked right away and all the containers started right away <3

The backup didn't work with the default directory but I created a folder on the DSM which worked without additional settings: `/volume1/docker/nextcloud/backup`
I will add this to the readme as well.

The last thing I don't understand is why the firewall blocks everything from starting and how I can set it up right?
These are my current settings:
![Screenshot 2023-01-19 at 12 53 03](https://user-images.githubusercontent.com/70434961/213435805-54636fda-d5ce-4b4f-8e57-0d8c725a1767.jpg)

Any Idea?

**Reply by szaimen** [reply]
> The last thing I don't understand is why the firewall blocks everything from starting and how I can set it up right?

Some of your firewall settings seem to block docker networks and docker nat from functioning correctly. Unfortunately I cannot help you to figure out how to fix this. I suppose you would need to google that.

**Reply by Fred-DTV** [reply]
Ok my bad, as I was able to connect to other docker containers I was assuming it would be a AIO issue, but you were right: I had to add special expceptions to the firewall for the docker network bond:
![Screenshot 2023-01-19 at 14 13 48](https://user-images.githubusercontent.com/70434961/213451940-8d3404d3-8f23-4876-934e-41e2aae9241c.jpg)


I will add this to the readme PR too :)

**Reply by szaimen** [reply]
> I will add this to the readme PR too :)

Great! I am looking forward the PR :)

**Reply by Fred-DTV** [reply]
Done :) Hopefully I didn't forget anything 

Thanks again for your help

**Comment by szaimen** [comment, marked as answer]
Will be documented with https://github.com/nextcloud/all-in-one/pull/1813
